### ISSUE TITLE
Add support for `-pg` to the LLVM back-end?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#23860]
When using the C back-end, `--ccflags -pg --ldflags -pg` can be used to generate a `gprof`-ready binary in Chapel.  However, when using the LLVM back-end, my understanding is that the Chapel compiler would need to do more manual work in creating LLVM IR to insert the profiling instructions itself—i.e., LLVM won't do it for us.  This issue asks for such support to be added, imagining that it may be a heavy lift.

**Comment by dlongnecke-cray** [comment]
Clang seems to have a `-pg` flag, so it may be possible to have it throw that for us: https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-pg

**Comment by mppf** [comment]
Yes but it's my understanding that clang's code generator (i.e. the part that translates C AST into LLVM IR) is the part that is adding the profiling calls. As a result, we would need to take the same steps in our own code generator, for it to work in general.

**Comment by dlongnecke-cray** [comment]
Ah drat, I hadn't looked into it further. Thanks Michael.
